androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/favorites/FavoritesViewModelTest.kt

```diff
@@ -0,0 +1,166 @@
+    val now = Instant.fromEpochMilliseconds(System.currentTimeMillis())
+    val route =
+        builder.route {
+            id = "route_1"
+            type = RouteType.LIGHT_RAIL
+            color = "FF0000"
+            directionNames = listOf("North", "South")
+            directionDestinations = listOf("Downtown", "Uptown")
+            longName = "Sample Route Long Name"
+            shortName = "Sample Route"
+            textColor = "000000"
+            lineId = "line_1"
+            routePatternIds = mutableListOf("pattern_1", "pattern_2")
+        }
+    val routePatternOne =
+        builder.routePattern(route) {
+            id = "pattern_1"
+            directionId = 0
+            name = "Sample Route Pattern"
+            routeId = "route_1"
+            representativeTripId = "trip_1"
+        }
+    val routePatternTwo =
+        builder.routePattern(route) {
+            id = "pattern_2"
+            directionId = 1
+            name = "Sample Route Pattern Two"
+            routeId = "route_1"
+            representativeTripId = "trip_1"
+        }
+    val stop1 =
+        builder.stop {
+            id = "stop_1"
+            name = "Sample Stop"
+            locationType = LocationType.STOP
+            latitude = 0.0
+            longitude = 0.0
+        }
+    val stop2 =
+        builder.stop {
+            id = "stop_2"
+            name = "Sample Stop 2"
+            locationType = LocationType.STOP
+            latitude = 0.0
+            longitude = 0.0
+        }
+
+    val stop3 =
+        builder.stop {
+            id = "stop_3"
+            name = "Sample Stop 3"
+            locationType = LocationType.STOP
+            latitude = 0.0
+            longitude = 0.0
+        }
+    val line =
+        builder.line {
+            id = "line_1"
+            color = "FF0000"
+            textColor = "FFFFFF"
+        }
+    val trip =
+        builder.trip {
+            id = "trip_1"
+            routeId = "route_1"
+            directionId = 0
+            headsign = "Sample Headsign"
+            routePatternId = "pattern_1"
+            stopIds = listOf(stop1.id, stop2.id, stop3.id)
+        }
+    val prediction =
+        builder.prediction {
+            id = "prediction_1"
+            revenue = true
+            stopId = "stop_1"
+            tripId = "trip_1"
+            routeId = "route_1"
+            stopSequence = 1
+            directionId = 0
+            arrivalTime = now.plus(1.minutes)
+            departureTime = now.plus(1.5.minutes)
+        }
+
+    private val globalResponse =
+        GlobalResponse(
+            builder,
+            mutableMapOf(
+                stop1.id to listOf(routePatternOne.id, routePatternTwo.id),
+                stop2.id to listOf(routePatternOne.id, routePatternTwo.id),
+            ),
+        )
+
+    val koinApplication = testKoinApplication(builder)
+
+    @Test
+    fun testFiltering() = runBlocking {
+        val favoritesVM = FavoritesViewModel()
```

Comment:
It looks like this test doesn’t require Android to be running at all, since the view model logic doesn’t rely on being run inside of Compose, which means this could be moved from `androidTest` to `test` to run on a cheaper instance in CI.